docker/Dockerfile

```diff
@@ -5,7 +5,7 @@
 # https://hub.docker.com/_/python
 # NOTE(willkg): stick with slim-bullseye which matches how we build the
 # stackwalker
-FROM --platform=linux/amd64 python:3.11.9-slim-bullseye@sha256:8850f5e6e8da9081a6d156252a11161aa22f04d6ed1723c57ca2d5a5d48132bc
+FROM --platform=linux/amd64 python:3.11.9-slim-bullseye@sha256:8850f5e6e8da9081a6d156252a11161aa22f04d6ed1723c57ca2d5a5d48132bc AS app_amd64
```

Comment:
fixes:
```
WARN: FromPlatformFlagConstDisallowed: FROM --platform flag should not use constant value "linux/amd64"
```
by using stage names as described in https://docs.docker.com/reference/build-checks/from-platform-flag-const-disallowed/